### PR TITLE
LCH-6356: Adding compatibility with guzzle > 7.5.  (3.x)

### PR DESCRIPTION
**Pull request by narendradesai**
Issue: https://backlog.acquia.com/browse/LCH-6356

### CODE REVIEW

File: src/ContentHubClient.php [hunk 1]
```diff
@@ -147,8 +151,16 @@ public function __call($method, $args) {
 
       $args = $this->addSearchCriteriaHeader($args);
 
-      return $this->httpClient->__call($method, $args);
+      if (\count($args) < 1) {
```
Comment (saxumVermes) [inline src/ContentHubClient.php]: Can we just remove __call? I don't think we need it at all. In related methods, such as get, put, patch, post can be passed to the guzzle http.

**Comment by narendradesai** [comment]
PR for CH3.0.x module: https://github.com/acquia/acquia_contenthub/pull/960

**Comment by narendradesai** [comment]
Setting minimum version of guzzle to 6.5
`drupal/core-recommended 9.4.11 requires guzzlehttp/guzzle ~6.5.8`

**Review by amangrover90 — COMMENTED** [review]
3.x module branch's tests are failing due to method declaration incompatibility. I think the same would happen for 2.x branch.

**Review by saxumVermes — APPROVED** [review]
Let's roll it!

**Pull request merged**